src/Common/CliFolderPathCalculatorCore.cs

```diff
@@ -28,17 +27,19 @@ static class CliFolderPathCalculatorCore
             return Path.Combine(homePath, DotnetProfileDirectoryName);
         }
 
-        public static string? GetDotnetHomePath(Func<string, string?>? getEnvironmentVariable = null)
```

Comment:
I think the owners of the code should comment on the fact that this removes the "mocking" of env. variables capability (I know that Environment.GetFolderPath would circumvent it anyway, but still). I don't know this code to tell if this is something we want, or just a one-off.

Reply:
We can keep it if necessary. I just noticed that in the entire repo, we have one place where we were passing `getEnvironmentVariable` and that place was also using the default value (on line 33). From that outlook, it seemed unused.